plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/GitHook.kt

```diff
@@ -151,19 +150,19 @@ open class KtlintInstallGitHookTask @Inject constructor(
     @get:Input
     internal val hookName: Property<String> = objectFactory.property(String::class.java)
 
-    @get:InputDirectory
-    internal val projectDir: DirectoryProperty = objectFactory.directoryProperty().apply {
-        set(projectLayout.projectDirectory)
+    @get:Input
+    internal val projectDir: Property<String> = objectFactory.property(String::class.java).apply {
+        set(projectLayout.projectDirectory.asFile.absolutePath)
     }
 
-    @get:InputDirectory
-    internal val rootDirectory: DirectoryProperty = objectFactory.directoryProperty().apply {
-        set(project.rootDir)
+    @get:Input
+    internal val rootDirectory: Property<String> = objectFactory.property(String::class.java).apply {
+        set(project.rootDir.absolutePath)
     }
 
     @TaskAction
     fun installHook() {
-        val repo = RepositoryBuilder().findGitDir(projectDir.get().asFile).setMustExist(false).build()
```

Comment:
I think we have some work to do in terms of properly making this task depend on the _git directory_ as an input. But i think this PR is fine for now and we can refine later